src/components/lib/histogram.js

```diff
@@ -15,12 +15,55 @@ function getXScale(data, width) {
     nice();
 }
 
+// Computes the average of an array of numbers. If the array is empty, returns 1.
+function avg(arr) {
+  return arr.length > 0 ? arr.reduce((a,b)=>a+b)/arr.length : 1
+}
+
+// Computes min(|a|,|b|)/max(|a|,|b|).
+function fractionLT1(a,b) {
+  return Math.min(Math.abs(a),Math.abs(b))/Math.max(Math.abs(a),Math.abs(b))
+}
+
+function filterData(inputData, cutOff) {
```

Comment:
this name doesn't mean much to me.  Maybe ``intelligentTruncate'' or something more specific, with a description of what exactly it is doing?